pkg/storage/accounts.go

```diff
@@ -42,61 +47,53 @@ func NewBalanceOperator(s string) (BalanceOperator, bool) {
 	return BalanceOperator(s), true
 }
 
-func SetBalanceFilter(v string) func(*AccountsQuery) {
-	return func(q *AccountsQuery) {
-		q.Params["balance"] = v
-	}
-}
+func NewAccountsQuery() *AccountsQuery {
 
-func SetBalanceOperatorFilter(v BalanceOperator) func(*AccountsQuery) {
-	return func(q *AccountsQuery) {
-		q.Params["balance_operator"] = v
+	return &AccountsQuery{
+		Limit: QueryDefaultLimit,
 	}
 }
 
-func NewAccountsQuery(qms ...[]AccQueryModifier) AccountsQuery {
-	q := AccountsQuery{
-		Limit:  QueryDefaultLimit,
-		Params: map[string]interface{}{},
+func (a *AccountsQuery) WithLimit(limit uint) *AccountsQuery {
+	if limit != 0 {
```

Comment:
I see what you mean, i put the validation inside the WithXXX methods because i'd rather split as much complexity as possible outisde of important functions (like the controller), but i don't have any strong feelings about this.

also, if we put the validation outside of the methods, the methods become only setters, and it kind of seemed strange to me ?

@gfyrag what do you think ?

Reply:
We will soon need to validate somewhere the limit param when the users will be able to set it themselves. If we keep this method like this and a user passes `&limit=-1` for example, the user won't be warned that the limit is incorrect.
It doesn't seem strange to me that those methods are only setters :)

Reply:
I also think we should validate the parameters.
But for this particular PR, it does not changes the behavior, so i think it is ok.

Reply:
Alright ! So i'll merge this one, and we need to think about how we want to handle validation soon ! let's maybe start a topic on slack about this